tests/gtest/avifsvttest.cc

```diff
@@ -0,0 +1,125 @@
+// Copyright 2025 Google LLC
+// SPDX-License-Identifier: BSD-2-Clause
+
+#include <vector>
+
+#include "avif/avif.h"
+#include "aviftest_helpers.h"
+#include "gtest/gtest.h"
+
+namespace avif {
+namespace {
+
+class SvtAv1Test : public testing::TestWithParam<std::tuple</*quality=*/int>> {
+};
+
+TEST_P(SvtAv1Test, EncodeDecodeStillImage) {
+  const int quality = std::get<0>(GetParam());
+
+  if (avifCodecName(AVIF_CODEC_CHOICE_SVT, AVIF_CODEC_FLAG_CAN_ENCODE) ==
+          nullptr ||
+      !testutil::Av1DecoderAvailable()) {
+    GTEST_SKIP() << "Codec unavailable, skip test.";
+  }
+
+  // AVIF_CODEC_CHOICE_SVT requires dimensions to be at least 64 pixels.
+  ImagePtr image =
+      testutil::CreateImage(/*width=*/64, /*height=*/64, /*depth=*/8,
+                            AVIF_PIXEL_FORMAT_YUV420, AVIF_PLANES_ALL);
+  ASSERT_NE(image, nullptr);
+  testutil::FillImageGradient(image.get());
+  if (quality == AVIF_QUALITY_LOSSLESS) {
+    image->matrixCoefficients = AVIF_MATRIX_COEFFICIENTS_IDENTITY;
+  }
+
+  EncoderPtr encoder(avifEncoderCreate());
+  ASSERT_NE(encoder, nullptr);
+  encoder->codecChoice = AVIF_CODEC_CHOICE_SVT;
+  encoder->quality = quality;
+  encoder->qualityAlpha = quality;
+  testutil::AvifRwData encoded;
+  ASSERT_EQ(avifEncoderWrite(encoder.get(), image.get(), &encoded),
+            AVIF_RESULT_OK);
+
+  DecoderPtr decoder(avifDecoderCreate());
+  ASSERT_NE(decoder, nullptr);
+  ImagePtr decoded(avifImageCreateEmpty());
+  ASSERT_NE(decoded, nullptr);
+  ASSERT_EQ(avifDecoderReadMemory(decoder.get(), decoded.get(), encoded.data,
+                                  encoded.size),
+            AVIF_RESULT_OK);
+
+  if (quality == AVIF_QUALITY_LOSSLESS) {
+    // Lossless is not supported in SVT-AV1 2.3.0.
```

Comment:
I assume we set `svt_config->qp` to 0 in this case. We should ask the SVT-AV1 team what happens if `svt_config->qp` is set to 0 in SVT-AV1 2.3.0.

Reply:
As shown by the test, it encodes something but with loss.

Reply:
Thanks!